pkgs/development/compilers/flutter/artifacts/fetch-artifacts.nix

```diff
@@ -41,6 +41,14 @@ runCommand "flutter-artifacts-${platform}"
     inherit platform;
   };
 } ''
-  mkdir -p "$out"
-  HOME="$NIX_BUILD_TOP" FLUTTER_CACHE_DIR="$out" flutter precache -v '--${platform}' ${builtins.concatStringsSep " " (map (p: "'--no-${p}'") (lib.remove platform platforms))}
+  export FLUTTER_ROOT="$NIX_BUILD_TOP"
+  lndir -silent '${flutter'}' "$FLUTTER_ROOT"
+  rm -rf "$FLUTTER_ROOT/bin/cache"
+  mkdir "$FLUTTER_ROOT/bin/cache"
+
+  HOME="$(mktemp -d)" flutter precache -v '--${platform}' ${builtins.concatStringsSep " " (map (p: "'--no-${p}'") (lib.remove platform platforms))}
+  rm -r "$FLUTTER_ROOT/bin/cache/lockfile"
```

Comment:
after `move-cache.patch`, this should be removed. (since the lockfile will be in ~/.cache)